src/python-common/ceph/deployment/inventory.py

```diff
@@ -16,7 +18,14 @@ def __init__(self, devices):
         self.devices = devices  # type: List[Device]
 
     def __eq__(self, other: Any) -> bool:
-        return self.to_json() == other.to_json()
+        if not isinstance(other, Devices):
+            return NotImplemented
```

Comment:
can we have this raise a more descriptive error here. NotImpletmented makes sense for a top level command but if we had a bug that caused us to compare a device to something else (only way this case would occur) I think we'd like something that tells us "Tried to compare device to x".

Reply:
@adk3798 you mean something like?

`       raise TypeError(f"Cannot compare instance of type {type(self)} with type {type(other)}")
`
returning a NotImplemented even when could mask some BUG (or make it harder to debug) it doesn't break anything.
Raising an exception could break some list operations [ref](https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror/44575926
)
If this is Ok then I can change the code.

Reply:
From what I can see in that thread returning NotImplemented here will get it to try other ways of comparing the objects. However, I don't think we want it to do that (it won't find any valid way for comparing a Device object to something else anyway I think) and really it should fail hard if this is ever hit (as we have a bug and want to catch it immediately). I can't think of a case where we would want this to be hit and not break the operation. If it gets that far something else went wrong and the operation should fail imo. So I'd go with the example line you provided here with the TypeError.